platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/AttributionDialogManager.java

```diff
@@ -0,0 +1,130 @@
+package com.mapbox.mapboxsdk.maps;
+
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.net.Uri;
+import android.support.annotation.NonNull;
+import android.support.v7.app.AlertDialog;
+import android.text.Html;
+import android.text.SpannableStringBuilder;
+import android.text.TextUtils;
+import android.text.style.URLSpan;
+import android.view.View;
+import android.widget.ArrayAdapter;
+
+import com.mapbox.mapboxsdk.R;
+import com.mapbox.mapboxsdk.style.sources.Source;
+import com.mapbox.services.android.telemetry.MapboxTelemetry;
+
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+
+class AttributionDialogManager implements View.OnClickListener, DialogInterface.OnClickListener {
+
+  private final Context context;
+  private final MapboxMap mapboxMap;
+  private String[] attributionKeys;
+  private HashMap<String, String> attributionMap;
+
+  AttributionDialogManager(@NonNull Context context, @NonNull MapboxMap mapboxMap) {
+    this.context = context;
+    this.mapboxMap = mapboxMap;
+  }
+
+  // Called when someone presses the attribution icon on the map
+  @Override
+  public void onClick(View view) {
+    rebuildAttributionMap();
+    showAttributionDialog();
+  }
+
+  private void rebuildAttributionMap() {
+    attributionMap = new LinkedHashMap<>();
+    for (Source source : mapboxMap.getSources()) {
+      parseAttribution(source.getAttribution());
+    }
+    addTelemetryEntryToAttributionMap();
+  }
+
+  private void parseAttribution(String attributionSource) {
+    if (!TextUtils.isEmpty(attributionSource)) {
+      SpannableStringBuilder htmlBuilder = (SpannableStringBuilder) Html.fromHtml(attributionSource);
```

Comment:
No external icon support is perfectly fine. I just wanted to double-check because the use of HTML in attribution has been problematic for GL JS and mapbox.js, mostly concerning XSS risk.